config/constants.go

```diff
@@ -18,8 +16,9 @@ const (
 )
 
 var (
-	// PublicHLSStoragePath is the directory we write public HLS files to for distribution.
-	PublicHLSStoragePath = filepath.Join(WebRoot, "hls")
 	// BackupDirectory is the directory we write backup files to.
 	BackupDirectory = filepath.Join(DataDirectory, "backup")
+
+	// HLSStoragePath is the directory HLS video is written to.
+	HLSStoragePath = filepath.Join(DataDirectory, "hls")
```

Comment:
I have a hard time understanding why the data directory is your preferred location for storage. Previously, it was clear that HLS files were served „with the webserver“ if they were in the `/webroot` folder. Now, the segments are in another folder which also serves anything that’s in it, but is not marked as „public for everyone to access“ as `/webroot`.

IMO, `HLSStoragePath` should always be `/webroot/hls`, even if S3 is activated, but I’m open for a debate here.

Reply:
It's because people have asked us to centralize where we write things so they can run Owncast from read-only locations and only have to map a single writable directory. If we work towards having `data` be the single place where we read and write user data then it simplifies concerns when people deploy.  We're not there yet as we're still writing images to `webroot`, but I'd like to move those too.  But going forward we should not be writing to `webroot`.
 
It also very much makes sense to simplify and always write files to the same location instead of the weird "sometimes here and sometimes there" logic we currently have. A single location where if public access is needed then a endpoint specifically built for that purpose should be used, not just throwing the files under `webroot`.  

And absolutely these files should not be publicly available if the configuration says video is not being served locally.

Reply:
Ack, thanks for clarifying.